auth/providers/azure/pop_tokenverifier.go

```diff
@@ -136,6 +143,28 @@ func (p *PoPTokenVerifier) ValidatePopToken(token string) (string, error) {
 		return "", errors.Errorf("Invalid token. 'u' claim is missing")
 	}
 
+	// Verify host 'nonce' claim
+	if nonce, ok := claims["nonce"]; ok {
+		if _, ok := nonce.(string); !ok {
+			return "", errors.Errorf("Invalid token. 'nonce' claim should be of string")
+		}
+	} else {
+		return "", errors.Errorf("Invalid token. 'nonce' claim is missing")
+	}
```

Comment:
Suggestion: Store nonce in a variable instead of indexing and casting for each usage.
```suggestion
	// Verify host 'nonce' claim
	var nonce string
	if nonceClaim, ok := claims["nonce"]; ok {
		if nonce, ok = nonceClaim.(string); !ok {
			return "", errors.Errorf("Invalid token. 'nonce' claim should be of string")
		}
	} else {
		return "", errors.Errorf("Invalid token. 'nonce' claim is missing")
	}
```